sled-agent/src/services.rs

```diff
@@ -634,19 +638,17 @@ impl ServiceManager {
                             },
                         ],
                         dropshot_internal: dropshot::ConfigDropshot {
-                            bind_address: SocketAddr::new(IpAddr::V6(*internal_ip), NEXUS_INTERNAL_PORT),
+                            bind_address: SocketAddr::new(
+                                IpAddr::V6(*internal_ip),
+                                NEXUS_INTERNAL_PORT,
+                            ),
                             request_body_max_bytes: 1048576,
                             ..Default::default()
                         },
                         subnet: Ipv6Subnet::<RACK_PREFIX>::new(
                             self.inner.underlay_address,
                         ),
-                        // TODO: Switch to inferring this URL by DNS.
-                        database: nexus_config::Database::FromUrl {
-                            url: PostgresConfigWithUrl::from_str(
-                                "postgresql://root@[fd00:1122:3344:0101::3]:32221/omicron?sslmode=disable"
-                            ).unwrap(),
-                        }
+                        database: nexus_config::Database::FromDns,
```

Comment:
As a nice perk, because we're setting the DNS record correctly, Nexus can connect to CRDB regardless of what IP it has.